bootstrap/kubeadm/api/v1alpha2/conversion.go

```diff
@@ -17,22 +17,36 @@ limitations under the License.
 package v1alpha2
 
 import (
-	"errors"
-
 	apiconversion "k8s.io/apimachinery/pkg/conversion"
 	kubeadmbootstrapv1alpha3 "sigs.k8s.io/cluster-api/bootstrap/kubeadm/api/v1alpha3"
+	utilconversion "sigs.k8s.io/cluster-api/util/conversion"
 	"sigs.k8s.io/controller-runtime/pkg/conversion"
 )
 
 // ConvertTo converts this KubeadmConfig to the Hub version (v1alpha3).
 func (src *KubeadmConfig) ConvertTo(dstRaw conversion.Hub) error { // nolint
 	dst := dstRaw.(*kubeadmbootstrapv1alpha3.KubeadmConfig)
+
+	// Manually restore data.
+	restored := &kubeadmbootstrapv1alpha3.KubeadmConfig{}
+	if err := utilconversion.UnmarshalData(src, restored); err == nil {
```

Comment:
This seems to ignore all errors, should we not ignore errors that are related to the json.Unmarshal call?